### ISSUE TITLE
New Dependencies.toml v2 is not picking up dependencies from old Dependencies.toml

### ISSUE THREAD

**Issue by Mirage20** [ballerina-platform/ballerina-lang#32793]
**Description:**

We have a project created from Alpha5 which contains old Dependencies.toml with few connector versions locked in. When we try to build the project with Beta3 it updates the Dependencies.toml to v2 and bumps the locked versions to the next available major version. 

**Steps to reproduce:**

1. Create a project with the following content

   *  `$ cat integration.bal`

      ```ballerina
      import ballerina/http;
      import ballerina/log;
      import ballerina/websub;
      import ballerinax/github.webhook;
      import ballerinax/googleapis.sheets as sheets;
      
      // GitHub configuration parameters
      @display {
          kind: "OAuthConfig",
          provider: "GitHub",
          label: "Set Up GitHub Connection"
      }
      configurable http:BearerTokenConfig & readonly gitHubTokenConfig = ?;
      
      @display {
          kind: "ConnectionField",
          connectionRef: "gitHubTokenConfig",
          provider: "GitHub",
          operationName: "getUserRepositoryList",
          label: "GitHub Repository URL"
      }
      configurable string gitHubRepoUrl = ?;
      
      @display {
          kind: "WebhookURL",
          label: "Set Up Callback URL for GitHub Webhook"
      }
      configurable string gitHubCallbackUrl = ?;
      
      // Google sheets configuration parameters
      @display {
          kind: "OAuthConfig",
          provider: "Google Sheets",
          label: "Set Up Google Sheets Connection"
      }
      configurable http:OAuth2RefreshTokenGrantConfig & readonly sheetOauthConfig = ?;
      
      @display {
          kind: "ConnectionField",
          connectionRef: "sheetOauthConfig",
          provider: "Google Sheets",
          operationName: "getAllSpreadsheets",
          label: "Spreadsheet Name"
      }
      configurable string spreadsheetId = ?;
      
      @display {
          kind: "ConnectionField",
          connectionRef: "sheetOauthConfig",
          provider: "Google Sheets",
          operationName: "getSheetList",
          label: "Worksheet Name"
      }
      configurable string worksheetName = ?;
      
      final string[] & readonly headerValues = [
          "Issue Link",
          "Issue Number",
          "Issue Title",
          "Issue Creator",
          "Issue Created At"
      ];
      
      const HEADINGS_ROW = 1;
      
      // Initialize the Github listener
      listener webhook:Listener githubListener = new (8090);
      
      @websub:SubscriberServiceConfig {
          target: [webhook:HUB, gitHubRepoUrl + "/events/*.json"],
          callback: gitHubCallbackUrl + "/subscriber",
          httpConfig: {auth: gitHubTokenConfig}
      }
      service /subscriber on githubListener {
          remote function onIssuesOpened(webhook:IssuesEvent event) returns error? {
              sheets:Client spreadsheetClient = check new ({auth: {token: ""}});
      
              sheets:Row headers = check spreadsheetClient->getRow(spreadsheetId, worksheetName, HEADINGS_ROW);
              if headers.values.length() == 0 {
                  check spreadsheetClient->appendRowToSheet(spreadsheetId, worksheetName, headerValues);
              }
      
              (int|string|decimal)[] values = [event.issue.html_url, event.issue.number, event.issue.title, event.issue.user.login, event.issue.created_at];
              check spreadsheetClient->appendRowToSheet(spreadsheetId, worksheetName, values);
              log:printInfo("New row appended successfully!");
          }
      }
      ```

    * $ cat Ballerina.toml 
      ```toml
      [package]
      org = "mirajabeysekara"
      name = "github_issue_to_gsheet"
      version = "0.1.0"
      
      [build-options]
      observabilityIncluded = true
      ```

    * $ cat Dependencies-old.toml 
      ```toml
      [[dependency]]
      org = "ballerinax"
      name = "github"
      version = "0.99.18"
      
      [[dependency]]
      org = "ballerinax"
      name = "googleapis.sheets"
      version = "0.99.12"
      ```
2. Build using the following command
```
cp Dependencies-old.toml Dependencies.toml &&  BALLERINA_STAGE_CENTRAL=true bal build
```

**Expected:**

Once the app build is finished, we expected to have v2 Dependencies.toml with `github` and `googleapis.sheets` versions locked to `0.99.20` and `0.99.13` respectively. (At the moment these are available in staging central)

**Actual:**

The v2 Dependencies.toml is generated with `github` and `googleapis.sheets` versions locked to `1.0.2` and `1.0.2`. 

This is the generated toml.

```toml
# AUTO-GENERATED FILE. DO NOT MODIFY.

# This file is auto-generated by Ballerina for managing dependency versions.
# It should not be modified by hand.

[ballerina]
dependencies-toml-version = "2"

[[package]]
org = "ballerina"
name = "auth"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "crypto"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.array"},
	{org = "ballerina", name = "lang.string"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "regex"}
]

[[package]]
org = "ballerina"
name = "cache"
version = "3.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "task"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "crypto"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "file"
version = "1.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "os"},
	{org = "ballerina", name = "regex"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "http"
version = "2.0.0"
transitive = false
dependencies = [
	{org = "ballerina", name = "auth"},
	{org = "ballerina", name = "cache"},
	{org = "ballerina", name = "crypto"},
	{org = "ballerina", name = "file"},
	{org = "ballerina", name = "io"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "jwt"},
	{org = "ballerina", name = "lang.array"},
	{org = "ballerina", name = "lang.decimal"},
	{org = "ballerina", name = "lang.int"},
	{org = "ballerina", name = "lang.runtime"},
	{org = "ballerina", name = "lang.string"},
	{org = "ballerina", name = "lang.value"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "mime"},
	{org = "ballerina", name = "oauth2"},
	{org = "ballerina", name = "observe"},
	{org = "ballerina", name = "regex"},
	{org = "ballerina", name = "time"},
	{org = "ballerina", name = "url"}
]
modules = [
	{org = "ballerina", packageName = "http", moduleName = "http"}
]

[[package]]
org = "ballerina"
name = "io"
version = "1.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.value"}
]

[[package]]
org = "ballerina"
name = "jballerina.java"
version = "0.0.0"
transitive = true

[[package]]
org = "ballerina"
name = "jwt"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "cache"},
	{org = "ballerina", name = "crypto"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.int"},
	{org = "ballerina", name = "lang.string"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "regex"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "lang.__internal"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.object"}
]

[[package]]
org = "ballerina"
name = "lang.array"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.__internal"}
]

[[package]]
org = "ballerina"
name = "lang.decimal"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "lang.int"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "lang.object"
version = "0.0.0"
transitive = true

[[package]]
org = "ballerina"
name = "lang.runtime"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.array"},
	{org = "ballerina", name = "lang.value"}
]

[[package]]
org = "ballerina"
name = "lang.string"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "lang.value"
version = "0.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "log"
version = "2.0.0"
transitive = false
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "observe"}
]
modules = [
	{org = "ballerina", packageName = "log", moduleName = "log"}
]

[[package]]
org = "ballerina"
name = "mime"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "io"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.int"}
]

[[package]]
org = "ballerina"
name = "oauth2"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "cache"},
	{org = "ballerina", name = "crypto"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "observe"
version = "0.9.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "os"
version = "1.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "random"
version = "1.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "regex"
version = "1.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "task"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "time"}
]

[[package]]
org = "ballerina"
name = "time"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "url"
version = "2.0.0"
transitive = true
dependencies = [
	{org = "ballerina", name = "jballerina.java"}
]

[[package]]
org = "ballerina"
name = "websub"
version = "2.0.0"
transitive = false
dependencies = [
	{org = "ballerina", name = "crypto"},
	{org = "ballerina", name = "http"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.string"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "mime"},
	{org = "ballerina", name = "random"},
	{org = "ballerina", name = "regex"},
	{org = "ballerina", name = "url"}
]
modules = [
	{org = "ballerina", packageName = "websub", moduleName = "websub"}
]

[[package]]
org = "ballerinai"
name = "observe"
version = "0.0.0"
transitive = false
dependencies = [
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "observe"}
]
modules = [
	{org = "ballerinai", packageName = "observe", moduleName = "observe"}
]

[[package]]
org = "ballerinax"
name = "github"
version = "1.0.2"
transitive = false
dependencies = [
	{org = "ballerina", name = "http"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.runtime"},
	{org = "ballerina", name = "lang.string"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "os"},
	{org = "ballerina", name = "regex"},
	{org = "ballerina", name = "websub"},
	{org = "ballerinai", name = "observe"}
]
modules = [
	{org = "ballerinax", packageName = "github", moduleName = "github"},
	{org = "ballerinax", packageName = "github", moduleName = "github.webhook"}
]

[[package]]
org = "ballerinax"
name = "googleapis.sheets"
version = "1.0.2"
transitive = false
dependencies = [
	{org = "ballerina", name = "http"},
	{org = "ballerina", name = "jballerina.java"},
	{org = "ballerina", name = "lang.array"},
	{org = "ballerina", name = "lang.int"},
	{org = "ballerina", name = "log"},
	{org = "ballerina", name = "os"},
	{org = "ballerina", name = "regex"}
]
modules = [
	{org = "ballerinax", packageName = "googleapis.sheets", moduleName = "googleapis.sheets"},
	{org = "ballerinax", packageName = "googleapis.sheets", moduleName = "googleapis.sheets.listener"},
	{org = "ballerinax", packageName = "googleapis.sheets", moduleName = "googleapis.sheets.sheets"}
]


```

**Affected Versions:**
Beta3 RC3

**Comment by Mirage20** [comment]
Following the above-mentioned steps will lead to a different error as below when executing the build command

```
$ cp Dependencies-old.toml Dependencies.toml &&  BALLERINA_STAGE_CENTRAL=true bal build 
Compiling source
        mirajabeysekara/github_issue_to_gsheet:0.1.0
error: compilation failed: Incompatible versions: ballerinax/googleapis.sheets. Version specified in Dependencies.toml: 0.99.12 and the version resolved from other dependencies: 1.0.2
```

There are few additional steps required in order to get to the originally mentioned state. 

1. Clear the Ballerina home `rm -rf ~/.ballerina/`
2. Pull the direct dependencies from the central to local

```
BALLERINA_STAGE_CENTRAL=true bal pull ballerinax/googleapis.sheets:1.0.2
BALLERINA_STAGE_CENTRAL=true bal pull ballerinax/github:1.0.2
BALLERINA_STAGE_CENTRAL=true bal pull ballerinax/github:0.99.20
BALLERINA_STAGE_CENTRAL=true bal pull ballerinax/googleapis.sheets:0.99.13 
```
3. Create the following file in the project target directory

    * `$ cat target/build`
    ```
    {
      "last_build_time": 1631865880234,
      "last_update_time": 1631862327338
    }
    ```
4. Run `cp Dependencies-old.toml Dependencies.toml &&  BALLERINA_STAGE_CENTRAL=true bal build`

**Comment by Mirage20** [comment]
Related: https://github.com/ballerina-platform/ballerina-lang/issues/32683

**Comment by Mirage20** [comment]
Looks like this issue is still there in Beta3 RC4

**Comment by azinneera** [comment]
The case here is as follows:

- **Dependencies.toml v1**: github `0.99.12`
- **Central**: github `0.99.20`, `2.0.0`

ATM, in the request sent to the Ballerina Central we only send the `org` and the `moduleName` and so the central resolves to `2.0.0` since this is the latest. This results in the `incompatible versions...` error. This happens only with v1 format of Dependencies.toml since we also have the modules list in v2 format which we use to find the package name.

As the fix, we need to send the list of the possible packages. The Central API also needs to change to accept this list and resolve with `MEDIUM` lock if there exist any possible packages. @sameerajayasoma WDYT?
